storage/remote/otlptranslator/prometheusremotewrite/metrics_to_prw.go

```diff
@@ -272,3 +278,52 @@ func (c *PrometheusConverter) addSample(sample *prompb.Sample, lbls []prompb.Lab
 	ts.Samples = append(ts.Samples, *sample)
 	return ts
 }
+
+func NewPromoteResourceAttributes(otlpCfg config.OTLPConfig) *PromoteResourceAttributes {
+	createAttr := func(attributes []string) map[string]struct{} {
+		attr := make(map[string]struct{}, len(attributes))
+		for _, s := range attributes {
+			attr[s] = struct{}{}
+		}
+		return attr
+	}
+
+	if otlpCfg.PromoteAllResourceAttributes {
+		return &PromoteResourceAttributes{
+			promoteAll: true,
+			attrs:      createAttr(otlpCfg.IgnoreResourceAttributes),
+		}
+	}
+
+	return &PromoteResourceAttributes{
+		promoteAll: false,
+		attrs:      createAttr(otlpCfg.PromoteResourceAttributes),
+	}
+}
+
+// promotedAttributes returns labels for promoted resourceAttributes.
+func (s *PromoteResourceAttributes) promotedAttributes(resourceAttributes pcommon.Map) []prompb.Label {
+	if s == nil {
+		return nil
+	}
```

Comment:
[Nit] For readability:
```suggestion
	}

```